ddc/developer-guide/quick-start.md

```diff
@@ -1,131 +1,157 @@
+    console.log(`New Bucket Id: ${bucketId}`);
+
+    console.log("========================= Store =========================");
+    // Store encrypted data
+    const secretPieceArray = new PieceArray(secretData, [topSecretDataTag]);
+    const secretUri = await ddcClientBob.store(bucketId, secretPieceArray, {encrypt: true, dekPath: subDekPath});
+    console.log(`Secret URI: ${secretUri}`)
+
+    // Store unencrypted data
+    const publicPieceArray = new PieceArray(publicData, [publicDataTag]);
+    const publicUri = await ddcClientBob.store(bucketId, publicPieceArray);
+    console.log(`Public URI: ${publicUri}`)
+
+
+    console.log("========================= Read data for Bob =========================");
+    // Read secret data without decryption
+    console.log("Read encrypted data for Bob:");
+    let pieceArray = await ddcClientBob.read(secretUri);
+    await readData(pieceArray);
+    console.log("=========================");
+
+    // Read secret decrypted data
+    console.log("Read decrypted data for Bob:");
+    pieceArray = await ddcClientBob.read(secretUri, {dekPath: subDekPath, decrypt: true});
+    await readData(pieceArray);
+    console.log("=========================");
+
+    // Read public data
+    console.log("Read public data for Bob:");
+    pieceArray = await ddcClientBob.read(publicUri);
+    await readData(pieceArray);
+    console.log("=========================");
+
+
+    console.log("========================= Share data for Alice =========================");
+    // Init Alice client
+    const ddcClientAlice = await DdcClient.buildAndConnect(MNEMONIC_ALICE, {
+        clusterAddress: cdnUrl,
+        smartContract: DEVNET
+    });
+    // Share DEK
+    await ddcClientBob.shareData(bucketId, mainDekPath, u8aToHex(ddcClientAlice.boxKeypair.publicKey));
+
+
+    console.log("========================= Read data for Alice =========================");
+    // Read public data
+    console.log("Read public data for Alice:");
+    pieceArray = await ddcClientAlice.read(publicUri);
+    await readData(pieceArray);
+    console.log("=========================");
+
+    // Read secret decrypted data
+    console.log("Read decrypted data for Alice:");
+    pieceArray = await ddcClientAlice.read(secretUri, {dekPath: mainDekPath, decrypt: true});
+    await readData(pieceArray);
+    console.log("=========================");
+    
+    console.log("========================= Disconnect =========================");
+    await ddcClientBob.disconnect();
+    await ddcClientAlice.disconnect();
+}
 
-    // Create Bucket
-    let bucketId;
-    {
-        log("Create a bucket...");
-        const tx = contract.tx.bucketCreate(txOptionsPay, '{"replication": 3}', CLUSTER_ID);
-        const result = await sendTx(account, tx);
-        bucketId = ddcBucket.findCreatedBucketId(result.contractEvents || []);
-        log("New BucketId", bucketId, "\n");
+const readData = async (pieceArray) => {
```

Comment:
@svalasovich Why do we not extract this function to a lower level library?

Reply:
Piece has async iterable method `dataReader` which returns bytes by chunks and for large files has lazy loading. Print in console is too specific method for usage.